docs/source/transaction_family_specifications/validator_registry_transaction_family.rst

```diff
@@ -0,0 +1,257 @@
+***************************************************
+Validator Registry Transaction Family Specification
+***************************************************
+
+Overview
+=========
+
+The validator registry transaction family provides a way to add new validators
+to the network.
+
+The validators information is used by poet_consensus to verify that validators
+trying to claim a block are following the block claiming policies of PoET
+consensus. For example, the Z policy will refuse blocks from a validator,
+if it has already claimed a disproportionate amount of blocks compared to the
+other validators on the network. The C policy says that a validator may not
+claim blocks, after being added to the registry or updating their information,
+until a certain number of blocks are claimed by other participants. The number
+of blocks claimed since signup can be found by checking the current block num
+against the block num that is stored in the validator info.  And finally the K
+policy, requires new signup information to be sent to the validator registry
+after it has claimed a maximum number of block. If the validator is not found
+in the validator registry the block will be rejected since there is no way to
+validate.
+
+Currently the number of blocks claimed by a specific validator is stored within
+the poet_enclave, not within the validator registry.
+
+
+State
+=====
+This section describes in detail how validator information, including
+identification and sign up data is stored and addressed using the validator
+registry transaction family.
+
+The following protocol buffers definition defines the validator info:
+
+.. code-block:: protobuf
+
+  message ValidatorInfo {
+    // Used to check if the validator's signup has been revoked.
+    string registered = 1;
+
+    // The name of the endpoint. It should be a human readable name.
+    string name = 2;
+
+    // The validator's public key (currently using signer_pubkey as this is
+    // stored in the transaction header)
+    string id = 3;
+
+    // This is the protocol buffer described below.
+    SignUpInfo signup_info = 4;
+
+    // The current chain head’s block number at the time of submission.
+    // This is used as a form to time to help with the enforcement of the C
+    // and K tests
+    int32 block_num = 5:
+  }
+
+The above ValidatorInfo includes the following protocol buffer:
+
+.. code-block:: protobuf
+
+  message SignUpInfo {
+    // Encoded public key corresponding to private key used by PoET to sign
+    // wait certificates.
+    string poet_public_key = 1;
+
+    // Information that can be used internally to verify the validity of
+    // the signup information. This is a JSON formatted string.
```

Comment:
While that will probably always be true for IAS, I think that at this level it should simply be considered an opaque buffer.